### PR TITLE
[TACHYON-1258] Handle empty remote writes

### PR DESCRIPTION
**Pull request by calvinjia**
https://tachyon.atlassian.net/browse/TACHYON-1258

### CODE REVIEW

File: common/src/main/java/tachyon/network/protocol/RPCBlockWriteRequest.java [hunk 1]
```diff
@@ -59,13 +59,10 @@ public static RPCBlockWriteRequest decode(ByteBuf in) {
...
     long offset = in.readLong();
     long length = in.readLong();
-    DataBuffer data = null;
-    if (length > 0) {
-      // TODO(hy): Look into accessing Netty ByteBuf directly, to avoid copying the data.
-      ByteBuffer buffer = ByteBuffer.allocate((int) length);
-      in.readBytes(buffer);
-      data = new DataByteBuffer(buffer, (int) length);
-    }
+    // TODO(hy): Look into accessing Netty ByteBuf directly, to avoid copying the data.
```
Comment (apc999) [inline common/src/main/java/tachyon/network/protocol/RPCBlockWriteRequest.java]: could you comment here that `length` can be zero and we still allocate the buffer and make the request, and also state the assumption we have here for `DataServerHandler`?

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: common/src/main/java/tachyon/network/protocol/RPCBlockWriteRequest.java (hunk 1 above)
Comment (calvinjia) [inline common/src/main/java/tachyon/network/protocol/RPCBlockWriteRequest.java]: Sure, updated.

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by gpang** [comment]
LGTM

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**